.github/workflows/tools_tests.yaml

```diff
@@ -58,8 +58,15 @@ jobs:
       run: pip install poetry==1.7.0
     - name: Installing python dependencies
       run: poetry install --no-cache --no-root
+    - uses: ./.github/workflows/setup-with-retry
+      with:
+        build_docker: 'false'
     - name: git LFS
       run: git lfs pull
+    - name: Get scons cache
```

Comment:
does the scons cache match between 20.04 and 24.04? i would think it needs its own cache

Reply:
there was some cache hits: https://github.com/commaai/openpilot/actions/runs/9556512327/job/26341924114?pr=32624#step:11:34

will check how much it match

Reply:
the Cython stuff for sure won't match